packages/core/src/PartialExecutionUtils/DirectedGraph.ts

```diff
@@ -67,6 +67,63 @@ export class DirectedGraph {
 		return this;
 	}
 
+	/**
+	 * Removes a node from the graph.
+	 *
+	 * By default it will also remove all connections that use that node and
+	 * return nothing.
+	 *
+	 * If you pass `{ reconnectConnections: true }` it will rewire all
+	 * connections making sure all parent nodes are connected to all child nodes
+	 * and return the new connections.
+	 */
+	removeNode(node: INode, options?: { reconnectConnections: true }): GraphConnection[];
+	removeNode(node: INode, options?: { reconnectConnections: false }): undefined;
+	removeNode(node: INode, { reconnectConnections = false } = {}): undefined | GraphConnection[] {
+		if (reconnectConnections) {
+			const incomingConnections = this.getDirectParents(node);
+			const outgoingConnections = this.getDirectChildren(node);
+
+			const newConnections: GraphConnection[] = [];
+
+			for (const incomingConnection of incomingConnections) {
+				for (const outgoingConnection of outgoingConnections) {
+					const newConnection = {
+						...incomingConnection,
+						to: outgoingConnection.to,
+						inputIndex: outgoingConnection.inputIndex,
+					};
```

Comment:
Does this work if the out connection has fewer inputs than what the deleted node has? E.g. a case like this

```
		//                xx
		//  ┌─────┐     ┌─────┐     ┌─────┐
		//  │     │o   o│     │o ┌►o│node2│
		//  │     │o─┐ o│     │o │  └─────┘
		//  │node0│o └►o│node1│o │  
		//  │     │o   o│     │o─┤
		//  │     │o   o│     │o 
		//  └─────┘     └─────┘     
```

I guess it should cause if I understood correctly every incoming is connected to every output?

Reply:
Good question. Let's try it out. I'll commit a test case for that.

Reply:
See [`c70b710` (#11024)](https://github.com/n8n-io/n8n/pull/11024/commits/c70b7100e9ccf241f6d844b9042b149c343cb90f)